torch/csrc/jit/codegen/cuda/lower_index.h

```diff
@@ -45,7 +46,7 @@ class TORCH_CUDA_CU_API IndexLowering : private OptOutConstDispatch {
   void generate(const std::vector<Expr*>& exprs);
 
   Val* lowerSrcIndex(Val* val, Val* dst) const;
-  Val* lowerDstIndex(Val* dst) const;
+  Val* lowerDstIndex(Val* dst, bool index_instruction_loop = true) const;
```

Comment:
Could this be simplified by having an initializer runtime function and just calling it instead of generating loops?

Reply:
If we can't do so, I still don't like this flag. All we need to know if if a certain Val appears in an MMA call site or the result buffer initialization. This can be determined by looking at the expression using these Vals, so we could pass the expression to `getConsumerIndex` and `getProducerIndex`. I think that's more generic and may be useful for other cases.

Reply:
Sure we can use a runtime function for initializing mma's, if that's preferrable. Would need to have a configurable one that works for all the mma macros. 

And also yes we could either pass the expr or inspect the loop nest to get this info.

Reply:
Either sounds good to me. I guess a runtime function would be easier.

Reply:
Initialization now done via runtime function. resolving.